content/docs/other-guides/virtual-dev/getting-started-multipass.md

```diff
@@ -4,117 +4,44 @@ description = "Quickly get Kubeflow running locally on native hypervisors"
+If you already have Ubuntu or Linux that supports [snaps](https://snapcraft.io/), you can easily install Kubernetes using [Microk8s](https://microk8s.io/). You can jump to the **Install Kubeflow using Microk8s** section below.
 
-  * quickly create disposable machine learning appliances
-  * leverage the same cloud images locally, reducing surprises when changing from development to staging to production environments in a multi-cloud strategy
-
-Here's a summary of the steps involved:
-
-1. Install the CLI
-2. Use the CLI to create a VM
-3. Enter the VM
-
-### 1. Install Multipass
-
-#### Mac OS X
-
-Install Multipass using the native Mac OS installer:
-
-* Download the latest \*-Darwin.pkg [here](https://github.com/CanonicalLtd/multipass/releases)
-* For more information and documentation, visit the project on [GitHub](https://github.com/CanonicalLtd/multipass)
-
-#### Linux
-
-Install Multipass using [snapcraft](https://snapcraft.io):
-
-```
-$ sudo snap install multipass --beta --classic
-```
-
-### 2. Create an Ubuntu virtual machine
-
-The following command creates a VM with 8GB of memory, 40GB of disk space, and 4 CPU. These are the minimum recommended settings. You are free to adjust them **higher** based on your host machine capabilities and workload requirements.
-```
-$ multipass launch bionic -n kubeflow -m 8G -d 40G -c 4
-```
-
-Note: If you need information on **resource utilization** in the VM, such as memory or disk space or CPU load, you can run ```multipass info kubeflow```
-
-### 3. Enter the VM
-
-```
-$ multipass shell kubeflow
-```
+If you don't have a Linux system already, or you would like to confine your Kubeflow to a disposable machine, then [Create a VM with Multipass](https://multipass.run/) first and then follow the instructions below. That will get you an Ubuntu machine that can be used to install Kubernetes and Kubeflow.
 
 ## Install Kubeflow using Microk8s
 
 Here's a summary of the steps involved:
 
 1. Set up Microk8s
-2. Set up Kubeflow
+2. Enable Kubeflow
 
 ### 1. Install and set up Microk8s
 
-This will install Microk8s if it doesn't already exist, and enable services that are useful for Kubeflow. Please inspect `setup-microk8s.sh` for more information.
-
 ```
-$ git clone https://github.com/canonical-labs/kubernetes-tools
-$ sudo kubernetes-tools/setup-microk8s.sh
+$ snap install microk8s --classic --channel=1.17/stable
+$ microk8s.status --wait-ready
+# enable common services
+$ microk8s.enable dns dashboard storage
 # If you have a GPU, run: `microk8s.enable gpu`
 ```
 
-If you would like access to the Kubernetes dashboard, please run this command:
-
-```
-kubernetes-tools/expose-dashboard.sh
-```
+### 2. Enable Kubeflow
 
-### 2. Install and set up Kubeflow
+$ microk8s.enable kubeflow
```

Comment:
I get an error when I run the `enable kubeflow` command, after following the instructions. Is there some command missing?

```
$ sudo /snap/bin/microk8s.enable kubeflow
Enabling dns...
Enabling storage...
Enabling dashboard...
Enabling ingress...
Enabling rbac...
Enabling juju...
Deploying Kubeflow...
ERROR unknown cloud "microk8s", please try "juju update-public-clouds"

Command '('microk8s-juju.wrapper', 'bootstrap', 'microk8s', 'uk8s')' returned non-zero exit status 1
Failed to enable kubeflow

$ sudo /snap/bin/microk8s.juju update-public-clouds
This operation can be applied to both a copy on this client and to the one on a controller.
No current controller was detected and there are no registered controllers on this client: either bootstrap one or register one.
Do you ONLY want to update public clouds on this client? (Y/n): y

Fetching latest public cloud list...
List of public clouds on this client is up to date, see `juju clouds --client`.

$ sudo /snap/bin/microk8s.enable kubeflow
Enabling dns...
Enabling storage...
Enabling dashboard...
Enabling ingress...
Enabling rbac...
Enabling juju...
Deploying Kubeflow...
ERROR unknown cloud "microk8s", please try "juju update-public-clouds"

Command '('microk8s-juju.wrapper', 'bootstrap', 'microk8s', 'uk8s')' returned non-zero exit status 1
Failed to enable kubeflow
```

Reply:
Here you go - the output from `snap info microk8s`:

```
name:      microk8s
summary:   Kubernetes for workstations and appliances
publisher: Canonical✓
contact:   https://github.com/ubuntu/microk8s
license:   unset
description: |
  MicroK8s is a small, fast, secure, single node Kubernetes that installs on just about
  any Linux box. Use it for offline development, prototyping, testing, or use it on a
  VM as a small, cheap, reliable k8s for CI/CD. It's also a great k8s for appliances -
  develop your IoT apps for k8s and deploy them to MicroK8s on your boxes.
commands:
  - microk8s.add-node
  - microk8s.cilium
  - microk8s.config
  - microk8s.ctr
  - microk8s.disable
  - microk8s.enable
  - microk8s.helm
  - microk8s.inspect
  - microk8s.istioctl
  - microk8s.join
  - microk8s.juju
  - microk8s.kubectl
  - microk8s.leave
  - microk8s.linkerd
  - microk8s.remove-node
  - microk8s.reset
  - microk8s.start
  - microk8s.status
  - microk8s.stop
services:
  microk8s.daemon-apiserver:          simple, enabled, active
  microk8s.daemon-apiserver-kicker:   simple, enabled, active
  microk8s.daemon-cluster-agent:      simple, enabled, active
  microk8s.daemon-containerd:         simple, enabled, active
  microk8s.daemon-controller-manager: simple, enabled, active
  microk8s.daemon-etcd:               simple, enabled, active
  microk8s.daemon-flanneld:           simple, enabled, inactive
  microk8s.daemon-kubelet:            simple, enabled, active
  microk8s.daemon-proxy:              simple, enabled, active
  microk8s.daemon-scheduler:          simple, enabled, active
snap-id:      EaXqgt1lyCaxKaQCU349mlodBkDCXRcg
tracking:     1.17/stable
refresh-date: 5 days ago, at 15:45 AEDT
channels:
  stable:           v1.17.0         2019-12-13 (1107) 179MB classic
  candidate:        v1.17.0         2019-12-10 (1107) 179MB classic
  beta:             v1.17.0         2019-12-10 (1107) 179MB classic
  edge:             v1.17.0         2020-01-10 (1145) 179MB classic
  dqlite/stable:    –                                       
  dqlite/candidate: –                                       
  dqlite/beta:      –                                       
  dqlite/edge:      v1.16.2         2019-11-07 (1038) 189MB classic
  1.18/stable:      –                                       
  1.18/candidate:   –                                       
  1.18/beta:        –                                       
  1.18/edge:        v1.18.0-alpha.1 2020-01-06 (1134) 179MB classic
  1.17/stable:      v1.17.0         2019-12-10 (1109) 179MB classic
  1.17/candidate:   v1.17.0         2019-12-10 (1109) 179MB classic
  1.17/beta:        v1.17.0         2019-12-10 (1109) 179MB classic
  1.17/edge:        v1.17.0         2020-01-08 (1140) 179MB classic
  1.16/stable:      v1.16.4         2020-01-13 (1117) 188MB classic
  1.16/candidate:   v1.16.4         2020-01-06 (1117) 188MB classic
  1.16/beta:        v1.16.4         2020-01-06 (1117) 188MB classic
  1.16/edge:        v1.16.4         2019-12-11 (1117) 188MB classic
  1.15/stable:      v1.15.7         2020-01-06 (1114) 171MB classic
  1.15/candidate:   v1.15.7         2019-12-17 (1114) 171MB classic
  1.15/beta:        v1.15.7         2019-12-17 (1114) 171MB classic
  1.15/edge:        v1.15.7         2019-12-11 (1114) 171MB classic
  1.14/stable:      v1.14.10        2020-01-06 (1120) 217MB classic
  1.14/candidate:   v1.14.10        2019-12-14 (1120) 217MB classic
  1.14/beta:        v1.14.10        2019-12-14 (1120) 217MB classic
  1.14/edge:        v1.14.10        2019-12-11 (1120) 217MB classic
  1.13/stable:      v1.13.6         2019-06-06  (581) 237MB classic
  1.13/candidate:   v1.13.6         2019-05-09  (581) 237MB classic
  1.13/beta:        v1.13.6         2019-05-09  (581) 237MB classic
  1.13/edge:        v1.13.7         2019-06-06  (625) 244MB classic
  1.12/stable:      v1.12.9         2019-06-06  (612) 259MB classic
  1.12/candidate:   v1.12.9         2019-06-04  (612) 259MB classic
  1.12/beta:        v1.12.9         2019-06-04  (612) 259MB classic
  1.12/edge:        v1.12.9         2019-05-28  (612) 259MB classic
  1.11/stable:      v1.11.10        2019-05-10  (557) 258MB classic
  1.11/candidate:   v1.11.10        2019-05-02  (557) 258MB classic
  1.11/beta:        v1.11.10        2019-05-02  (557) 258MB classic
  1.11/edge:        v1.11.10        2019-05-01  (557) 258MB classic
  1.10/stable:      v1.10.13        2019-04-22  (546) 222MB classic
  1.10/candidate:   v1.10.13        2019-04-22  (546) 222MB classic
  1.10/beta:        v1.10.13        2019-04-22  (546) 222MB classic
  1.10/edge:        v1.10.13        2019-04-22  (546) 222MB classic
installed:          v1.17.0                    (1109) 179MB classic
```

Reply:
Looks like the issue is that the fix was merged into master, but hasn't yet been pushed out into the latest snap. It will be coming with the next patch release. In the meantime, can you try `sudo snap switch microk8s --channel=edge && sudo snap refresh`, then `microk8s.enable kubeflow`? Getting that fix pushed out will probably block this PR from getting merged, but it would be good to check that that version is working for you when it does get released.

Reply:
I've tried it, but alas no luck:

```
$ sudo snap switch microk8s --channel=edge && sudo snap refresh
[sudo] password for sarahmaddox: 
"microk8s" switched to the "edge" channel

microk8s (edge) v1.17.1 from Canonical✓ refreshed

$ sudo /snap/bin/microk8s.enable kubeflow
Enabling dns...
Enabling storage...
Enabling dashboard...
Enabling ingress...
Enabling rbac...
Enabling juju...
Deploying Kubeflow...
ERROR unknown cloud "microk8s", please try "juju update-public-clouds"

Command '('microk8s-juju.wrapper', 'bootstrap', 'microk8s', 'uk8s')' returned non-zero exit status 1
Failed to enable kubeflow
```

Reply:
This is an issue with a cached binary/dir that aren't getting removed. Can you run these two commands, and try again?

    sudo rm -rf /var/snap/microk8s/*/juju
    sudo rm -rf /var/snap/microk8s/*/bin/juju

If you're on edge, you can also run `microk8s.disable juju`, which is equivalent. This command should also be hitting stable shortly.

The nuclear option is also to just completely remove the snap and reinstall it. That shouldn't be necessary, but if none of the above works, that will fix it. If you have to resort to that, I'd like to hear about it.

Reply:
Thanks @ammarn911 I'll let this go through, as I'm not the best person to test this update. I'm still getting the same error, but have run out of bandwidth here. I'll go with your testing of the fix.